common/src/main/java/io/pravega/common/util/ByteArraySegment.java

```diff
@@ -169,6 +171,45 @@ public void copyTo(OutputStream stream) throws IOException {
         stream.write(this.array, this.startOffset, this.length);
     }
 
+    @Override
+    @SneakyThrows(IOException.class)
+    public boolean equals(BufferView other) {
+        if (this.length != other.getLength()) {
+            return false;
+        } else if (other instanceof ArrayView) {
+            return equals((ArrayView) other);
+        }
+
+        InputStream otherReader = other.getReader();
+        for (int i = 0; i < this.length; i++) {
+            if ((byte) otherReader.read() != this.array[this.startOffset + i]) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean equals(ArrayView other) {
+        if (this.length != other.getLength()) {
+            return false;
+        }
+
+        byte[] otherArray = other.array();
+        int otherOffset = other.arrayOffset();
+        for (int i = 0; i < this.length; i++) {
```

Comment:
java.util.Arrays has HotSpot  intrinsified methods for array comparisons.

Reply:
They do, but they only compare whole arrays. I have array slices, so those methods aren't very helpful. Looking at them, they do exactly what I'm doing (byte-by-byte comparison) so not much of an improvement there.